esp/esp/themes/controllers.py

```diff
@@ -378,7 +378,7 @@ def get_file_summaries(self, dir):
                 result += self.get_file_summaries(full_filename)
             else:
                 file_data = open(full_filename, 'rb').read()
-                result.append((full_filename, os.path.getsize(full_filename), hashlib.sha1(file_data).hexdigest()))
+                result.append((full_filename, os.path.getsize(full_filename), hashlib.sha1(file_data.encode('UTF-8')).hexdigest()))
```

Comment:
This file_data is a sequence of bytes (because it's read with 'rb' ), not a string. It should not be encoded with UTF-8. Python 3 will give an error.  You can just pass the file_data directly to the sha1 function.